# Sabermetrics: The Air Quality Factor

In this Jupyter notebook I will explore the relationship between baseball performance and atmospheric air quality. The first step in this project is to import the multiple data sets with the goal of building a shared data frame so that the data can be easily compared using Pandas.


In [24]:
import numpy as np
import pandas as pd
import matplotlib.pylab as plt
%matplotlib inline

#Set filepaths for baseball data and atmospheric data
team_path = "C:\Users\Nathan\Documents\GitHub\SabermetricsProject\Data\Teams.csv"
epa_yearly_path = "C:\Users\Nathan\Documents\GitHub\SabermetricsProject\Data\epa_air_quality_annual_summary.csv"

team_DF = pd.read_csv(team_path)
epa_yearly_DF = pd.read_csv(epa_yearly_path)

#Rename columns to make table joins easier down the line
team_DF.rename(columns={"yearID":"year"},inplace=True)


#Print various stats about the data and dataframe
print("Original dataframe size = %s" %epa_yearly_DF.size)
print("Memory usage for EPA yearly summary = %s" % epa_yearly_DF.memory_usage)



Original dataframe size = 112129050
Memory usage for EPA yearly summary = <bound method DataFrame.memory_usage of         state_code  county_code  site_num  parameter_code  poc   latitude  \
0               20           57         1           82134    1  37.771964   
1               18          163        12           42401    1  38.021727   
2               40           31      9000           88152    1  34.732404   
3               32            3       561           42601    1  36.163959   
4               13          127      3001           12154    1  31.185187   
5               45           45         8           88313    1  34.838814   
6               06           99         2           83152    5  37.648819   
7               06           43         3           88342    1  37.713250   
8               42          101         4           45201    2  40.008889   
9               06           41         1           43815    3  37.972310   
10              50            3      80

Now that the data has been imported, the next  step is to clean the data by removing unneeded information and then  try to determine a shared key/column so that the dataframes can be merged. The epa readings contained in the dataframe were collected at a variety of cities within the US, many without any professional baseball team. There also may be cities that have atmospheric readings but no MLB team. Therefore I will identify the cities contained within both datasets and use these for my analysis. In addition, a new column within the team dataframe needs to be created that contains the city the team is located in. This can be derived by stripping the city name off the front of the team organization name.

In [ ]:
#Create a list of distinct cities where readings were collected
distinct_cities = epa_yearly_DF.city_name.unique()
MLB_cities = ["Baltimore","Boston","Chicago","Cleveland","Detroit","Houston","Kansas City","Anaheim",
             "Minneapolis","New York","Oakland","Seattle","St. Petersburg","Arlington","Phoenix",
             "Atlanta","Cincinnati","Denver","Los Angeles","Miami","Milwaukee","Philadelphia",
             "Pittsburgh","San Diego","San Francisco","St. Louis","Washington","St. Paul","Newark",]

shared_cities = [x for x in distinct_cities if x in MLB_cities]
epa_yearly_DF = epa_yearly_DF[epa_yearly_DF['city_name'].isin(shared_cities)]

print("Reduced dataframe size by %s" %epa_yearly_DF.size)

#Create a dictionary mapping teams to cities
team_cities = team_DF.name.unique()
team_to_city = {}
for words in team_cities:
    wordsList = words.split(" ")
    if wordsList[0]=="Kansas" or wordsList[0]=="San" or wordsList[0]=="Los" or wordsList[0]=="New" or wordsList[0]=="St.":
        team_to_city[words] = wordsList[0]+" "+wordsList[1]
    else:
        team_to_city[words] = wordsList[0]

#Add new city_name column to team dataframe to act as key for joining
team_DF["city_name"] = "default"

for index,row in team_DF.iterrows():
    if row["name"] in team_to_city:
        team_DF.set_value(index,"city_name",team_to_city[row["name"]])

merged_DF = pd.merge(team_DF,epa_yearly_DF,on=["year","city_name"])

plt.matshow(merged_DF.corr())

The correlation matrix above is way too large and is a mess. Therefore I will seperate the team columns into focus areas to try to make the important correlations more apparent.

In [7]:
team_col = ["W","H","2B","3B","ERA","FP","HR","SO","SF","E","teamID","year","city_name"]
stats_team_DF = team_DF[team_col]

merged_DF = pd.merge(stats_team_DF,epa_yearly_DF,on=["year","city_name"])

s = merged_DF.corr().unstack()
so = s.sort_values(kind="quicksort",ascending=False)

focus = ['W','ERA',"H","2B","3B","FP","HR","SO","E","SF"]
merged_DF.corr().filter(focus).drop(focus)

W       ERA         H        2B  \
year                              0.103986  0.016280  0.079861  0.323241   
county_code                      -0.028952  0.039561  0.015973 -0.009726   
site_num                          0.046779 -0.114188 -0.009427 -0.059859   
parameter_code                    0.081462  0.056529  0.108225  0.201229   
poc                               0.097523 -0.067067  0.070567  0.120021   
latitude                          0.000637  0.151073  0.111936  0.083421   
longitude                         0.059715  0.116239  0.086355  0.168792   
observation_count                -0.035513 -0.030333 -0.041880 -0.065306   
observation_percent               0.037524  0.027401  0.068226  0.116529   
valid_day_count                  -0.020782 -0.038420 -0.024510 -0.044575   
required_day_count               -0.013588 -0.057376 -0.041795 -0.081868   
exceptional_data_count           -0.010452  0.008518  0.002322  0.003845   
null_data_count                   0.010404 -0.015411 -0.000470  0.008860   
primary_exceedance_count         -0.001707 -0.118215 -0.051210 -0.147512   
secondary_exceedance_count        0.021586 -0.104769 -0.036386 -0.112681   
num_obs_below_mdl                      NaN       NaN       NaN       NaN   
arithmetic_mean                   0.015315  0.004471  0.016214  0.010907   
arithmetic_standard_dev           0.002556 -0.005000 -0.002835 -0.010280   
first_max_value                   0.010699 -0.004696  0.003507 -0.002396   
second_max_value                  0.010528 -0.004430  0.003640 -0.002816   
third_max_value                   0.010569 -0.004255  0.003695 -0.002691   
fourth_max_value                  0.010835 -0.004079  0.004073 -0.002258   
first_max_non_overlapping_value  -0.103300 -0.203959 -0.190111 -0.437768   
second_max_non_overlapping_value -0.097750 -0.209056 -0.186159 -0.435999   
ninety_nine_percentile            0.007695 -0.002252  0.003244 -0.003626   
ninety_eight_percentile           0.007308 -0.001570  0.003662 -0.003271   
ninety_five_percentile            0.007369 -0.000974  0.004771 -0.002192   
ninety_percentile                 0.008400 -0.000022  0.006869  0.000115   
seventy_five_percentile           0.012110  0.002671  0.012488  0.006628   
fifty_percentile                  0.017435  0.006277  0.019752  0.015284   
ten_percentile                    0.021724  0.010989  0.026586  0.025260   

                                        3B        FP        HR        SO  \
year                             -0.251888  0.559977  0.295951  0.737000   
county_code                       0.054882  0.047573 -0.039420 -0.113179   
site_num                          0.004963  0.054025 -0.097945  0.025515   
parameter_code                   -0.144662  0.202238  0.173891  0.235730   
poc                              -0.054802  0.156079  0.062519  0.127196   
latitude                         -0.044560  0.018860  0.155795 -0.034152   
longitude                        -0.062505  0.001550  0.206026  0.005677   
observation_count                 0.034678 -0.066843 -0.073474 -0.042649   
observation_percent              -0.055774  0.097972  0.066071  0.070629   
valid_day_count                   0.019473 -0.047955 -0.067689 -0.033792   
required_day_count                0.026246 -0.074257 -0.077727 -0.043780   
exceptional_data_count            0.004097  0.004129 -0.007738  0.002621   
null_data_count                  -0.025373  0.016994  0.003178  0.034955   
primary_exceedance_count          0.064173 -0.172860 -0.123512 -0.181526   
secondary_exceedance_count        0.046165 -0.145312 -0.085621 -0.141541   
num_obs_below_mdl                      NaN       NaN       NaN       NaN   
arithmetic_mean                  -0.014682  0.000246  0.020835  0.002404   
arithmetic_standard_dev          -0.005383 -0.018596 -0.002668 -0.013231   
first_max_value                  -0.009417 -0.011431  0.006691 -0.008086   
second_max_value                 -0.010467 -0.011778  0.006918 -0.008326   
third_max_va

From the table above we can see that there are not any significant correlations between the datasets. In the comparison above all different atmospheric readings are being lumped together and treated as the same. With over 1000 unique reading types it does not really make sense to keep them lumped together. Lets see if there are any correlations between a few specific reading types.

In [9]:
sulfurdioxide_DF = merged_DF[merged_DF["parameter_name"] == "Sulfur dioxide"]

focus = ['W','ERA',"H","2B","3B","FP","HR","SO","E","SF"]
sulfurdioxide_DF.corr().filter(focus).drop(focus)

W       ERA         H        2B  \
year                              0.109469  0.258738  0.186516  0.499711   
county_code                      -0.000669 -0.023015 -0.041185 -0.067368   
site_num                          0.021713 -0.021419  0.020168 -0.003732   
parameter_code                         NaN       NaN       NaN       NaN   
poc                               0.006296 -0.075081 -0.044872 -0.059233   
latitude                         -0.014117  0.211739  0.100779  0.075642   
longitude                         0.087707  0.045141  0.078181  0.112335   
observation_count                 0.017925 -0.026101 -0.008054  0.025688   
observation_percent               0.008425  0.104744  0.077451  0.123333   
valid_day_count                   0.013075  0.104908  0.083594  0.134422   
required_day_count               -0.001082 -0.010059  0.004508 -0.006887   
exceptional_data_count            0.001281 -0.019328  0.002345 -0.007062   
null_data_count                   0.021219 -0.021114  0.005948  0.031497   
primary_exceedance_count         -0.074707 -0.098993 -0.067524 -0.161947   
secondary_exceedance_count       -0.007149 -0.061661  0.011558 -0.022929   
num_obs_below_mdl                      NaN       NaN       NaN       NaN   
arithmetic_mean                  -0.049889 -0.149080 -0.088319 -0.244911   
arithmetic_standard_dev          -0.088233 -0.127112 -0.091279 -0.242534   
first_max_value                  -0.101870 -0.091821 -0.076468 -0.206303   
second_max_value                 -0.092068 -0.105477 -0.072086 -0.208180   
third_max_value                  -0.091647 -0.110209 -0.075285 -0.212585   
fourth_max_value                 -0.090531 -0.113325 -0.076878 -0.214988   
first_max_non_overlapping_value        NaN       NaN       NaN       NaN   
second_max_non_overlapping_value       NaN       NaN       NaN       NaN   
ninety_nine_percentile           -0.085018 -0.122863 -0.086393 -0.229044   
ninety_eight_percentile          -0.077980 -0.129816 -0.088128 -0.237713   
ninety_five_percentile           -0.070820 -0.137216 -0.090079 -0.245271   
ninety_percentile                -0.062503 -0.143714 -0.091499 -0.250501   
seventy_five_percentile          -0.048176 -0.152221 -0.090678 -0.250871   
fifty_percentile                 -0.034021 -0.149951 -0.084539 -0.236156   
ten_percentile                    0.008308 -0.126843 -0.046370 -0.162412   

                                        3B        FP        HR        SO  \
year                             -0.281702  0.621925  0.434604  0.756991   
county_code                       0.094357  0.084203 -0.070878 -0.016798   
site_num                         -0.047422  0.048280 -0.005615  0.057000   
parameter_code                         NaN       NaN       NaN       NaN   
poc                              -0.011359  0.028667 -0.042682  0.062222   
latitude                         -0.038463 -0.051219  0.165280 -0.144869   
longitude                        -0.089095 -0.055710  0.126426 -0.093834   
observation_count                -0.043257  0.100419  0.027575  0.142889   
observation_percent              -0.076080  0.047431  0.041109  0.093668   
valid_day_count                  -0.083268  0.048842  0.044147  0.102567   
required_day_count               -0.004987 -0.014664  0.001547 -0.001443   
exceptional_data_count           -0.008171 -0.012387  0.008667 -0.002761   
null_data_count                  -0.053757  0.079251  0.025380  0.106580   
primary_exceedance_count          0.099893 -0.169302 -0.183527 -0.211285   
secondary_exceedance_count        0.018467 -0.015210 -0.050065 -0.030291   
num_obs_below_mdl                      NaN       NaN       NaN       NaN   
arithmetic_mean                   0.077502 -0.316838 -0.213752 -0.384775   
arithmetic_standard_dev           0.137845 -0.300444 -0.242455 -0.373236   
first_max_value                   0.148210 -0.271634 -0.231408 -0.322320   
second_max_value                  0.141789 -0.271675 -0.234044 -0.329305   
third_max_va

After isolating atmospheric records for Sulfur Dioxide we can start seeing some meaningful results Notably there is a negative correlation (-0.384775) between the average sulfur dioxide concentrations and strike out rates. There is also a noteable negative correlation (-0.316838) with fielding percentage. Next we will examine the correlation of ozone.

In [17]:
ozone_DF = merged_DF[merged_DF["parameter_name"] == "Ozone"]
focus = ['W','ERA',"H","2B","3B","FP","HR","SO","E","SF"]
ozone_DF.corr().filter(focus).drop(focus)

W       ERA         H        2B  \
year                              0.043442  0.245016  0.120264  0.468783   
county_code                       0.008075 -0.071492 -0.021490 -0.018899   
site_num                          0.026146 -0.097912 -0.015898 -0.029748   
parameter_code                         NaN       NaN       NaN       NaN   
poc                              -0.027954 -0.034971 -0.045040 -0.033858   
latitude                         -0.000818  0.199562  0.096378  0.005622   
longitude                         0.047024  0.093769  0.073166  0.087265   
observation_count                 0.022104 -0.060791 -0.024963  0.040364   
observation_percent              -0.024228  0.118383  0.031525  0.111366   
valid_day_count                  -0.025184 -0.074299 -0.069324  0.027533   
required_day_count               -0.000595 -0.190893 -0.109487 -0.042810   
exceptional_data_count           -0.006882  0.043084  0.020649  0.021192   
null_data_count                   0.047414 -0.007162  0.028511  0.058681   
primary_exceedance_count          0.046597 -0.200037 -0.041931 -0.176287   
secondary_exceedance_count        0.046597 -0.200037 -0.041931 -0.176287   
num_obs_below_mdl                      NaN       NaN       NaN       NaN   
arithmetic_mean                   0.009471 -0.026386  0.020619 -0.029368   
arithmetic_standard_dev           0.035749 -0.219890 -0.050887 -0.245832   
first_max_value                   0.028811 -0.221974 -0.066079 -0.210993   
second_max_value                  0.022744 -0.200416 -0.056197 -0.195844   
third_max_value                   0.027606 -0.192679 -0.048657 -0.184581   
fourth_max_value                  0.028687 -0.186224 -0.041624 -0.176283   
first_max_non_overlapping_value        NaN       NaN       NaN       NaN   
second_max_non_overlapping_value       NaN       NaN       NaN       NaN   
ninety_nine_percentile            0.025558 -0.196895 -0.049703 -0.207917   
ninety_eight_percentile           0.027644 -0.185119 -0.040262 -0.195520   
ninety_five_percentile            0.023421 -0.158766 -0.028177 -0.175024   
ninety_percentile                 0.023788 -0.133013 -0.012884 -0.150152   
seventy_five_percentile           0.015224 -0.061471  0.008581 -0.082163   
fifty_percentile                  0.014743  0.031054  0.054822  0.035116   
ten_percentile                    0.012173  0.096025  0.066401  0.133119   

                                        3B        FP        HR        SO  \
year                             -0.391729  0.598138  0.469250  0.771023   
county_code                       0.099481  0.058192 -0.121628 -0.024318   
site_num                         -0.003814  0.068031 -0.053369  0.085574   
parameter_code                         NaN       NaN       NaN       NaN   
poc                               0.016785 -0.040480 -0.018841 -0.004904   
latitude                         -0.024201 -0.037740  0.147310 -0.158937   
longitude                        -0.071611  0.020160  0.137662 -0.047973   
observation_count                -0.055825  0.116708 -0.003076  0.140430   
observation_percent              -0.090943  0.128980  0.068780  0.126141   
valid_day_count                  -0.070236  0.089132 -0.044622  0.205711   
required_day_count               -0.007552  0.008853 -0.103585  0.160493   
exceptional_data_count           -0.018694  0.020176  0.046136  0.010939   
null_data_count                  -0.079664  0.065000  0.052748  0.058579   
primary_exceedance_count          0.095240 -0.252445 -0.171485 -0.237232   
secondary_exceedance_count        0.095240 -0.252445 -0.171485 -0.237232   
num_obs_below_mdl                      NaN       NaN       NaN       NaN   
arithmetic_mean                   0.046110 -0.067105 -0.053083 -0.059339   
arithmetic_standard_dev           0.192855 -0.359164 -0.242634 -0.406394   
first_max_value                   0.175881 -0.298587 -0.242086 -0.315320   
second_max_value                  0.165349 -0.289656 -0.229616 -0.301086   
third_max_va

In [19]:
carbonmonoxide_DF = merged_DF[merged_DF["parameter_name"] == "Carbon monoxide"]
focus = ['W','ERA',"H","2B","3B","FP","HR","SO","E","SF"]
carbonmonoxide_DF.corr().filter(focus).drop(focus)

W       ERA         H        2B  \
year                              0.125714  0.210088  0.180349  0.486171   
county_code                       0.009485 -0.015375 -0.014391 -0.060819   
site_num                          0.058743 -0.105751  0.002469 -0.004435   
parameter_code                         NaN       NaN       NaN       NaN   
poc                               0.038577 -0.062486 -0.002206  0.007796   
latitude                         -0.037565  0.225975  0.092205  0.087821   
longitude                         0.083551  0.022315  0.068437  0.115601   
observation_count                -0.003402  0.070038  0.034634  0.069158   
observation_percent              -0.004174  0.069946  0.033718  0.068604   
valid_day_count                  -0.000488  0.078181  0.041416  0.084306   
required_day_count                0.074957  0.005088  0.149254  0.043432   
exceptional_data_count           -0.020545  0.014342 -0.007527  0.014878   
null_data_count                   0.080698  0.012670  0.063001  0.144038   
primary_exceedance_count          0.017786 -0.078944 -0.037720 -0.087775   
secondary_exceedance_count        0.017786 -0.078944 -0.037720 -0.087775   
num_obs_below_mdl                      NaN       NaN       NaN       NaN   
arithmetic_mean                  -0.043336 -0.188325 -0.126310 -0.341340   
arithmetic_standard_dev          -0.094312 -0.215140 -0.174973 -0.424677   
first_max_value                  -0.096402 -0.174765 -0.164325 -0.388242   
second_max_value                 -0.100778 -0.188498 -0.174584 -0.412398   
third_max_value                  -0.101761 -0.194526 -0.177701 -0.419492   
fourth_max_value                 -0.099483 -0.196097 -0.175800 -0.421639   
first_max_non_overlapping_value  -0.103300 -0.203959 -0.190111 -0.437768   
second_max_non_overlapping_value -0.097750 -0.209056 -0.186159 -0.435999   
ninety_nine_percentile           -0.086915 -0.221520 -0.177647 -0.430216   
ninety_eight_percentile          -0.081982 -0.217106 -0.171407 -0.419257   
ninety_five_percentile           -0.073307 -0.207608 -0.156418 -0.395808   
ninety_percentile                -0.062914 -0.197952 -0.142115 -0.371274   
seventy_five_percentile          -0.049478 -0.183736 -0.124566 -0.336666   
fifty_percentile                 -0.030997 -0.170960 -0.108861 -0.305297   
ten_percentile                    0.012624 -0.143964 -0.073088 -0.219955   

                                        3B        FP        HR        SO  \
year                             -0.280398  0.617733  0.419289  0.758022   
county_code                       0.048399  0.108337 -0.043104 -0.046530   
site_num                         -0.016635  0.107963 -0.035562  0.137051   
parameter_code                         NaN       NaN       NaN       NaN   
poc                              -0.019910  0.068829  0.002161  0.064157   
latitude                         -0.052102 -0.034852  0.173388 -0.157465   
longitude                        -0.041173  0.024020  0.092970 -0.021894   
observation_count                -0.084689  0.047003  0.025447  0.044290   
observation_percent              -0.084548  0.046905  0.024562  0.043868   
valid_day_count                  -0.091745  0.060854  0.035030  0.064613   
required_day_count                0.046230 -0.011120  0.036682  0.025834   
exceptional_data_count           -0.019984  0.044601  0.006631  0.046200   
null_data_count                  -0.094415  0.152022  0.122157  0.177644   
primary_exceedance_count         -0.011679 -0.087643 -0.037987 -0.100650   
secondary_exceedance_count       -0.011679 -0.087643 -0.037987 -0.100650   
num_obs_below_mdl                      NaN       NaN       NaN       NaN   
arithmetic_mean                   0.029856 -0.412036 -0.214139 -0.471309   
arithmetic_standard_dev           0.118202 -0.467425 -0.307584 -0.542857   
first_max_value                   0.118547 -0.433266 -0.280711 -0.520510   
second_max_value                  0.126261 -0.457780 -0.297955 -0.541170   
third_max_va

In [20]:
particulate_DF = merged_DF[merged_DF["parameter_name"] == "Suspended particulate (TSP)"]
particulate_DF.corr().filter(focus).drop(focus)

W       ERA         H        2B  \
year                              0.139431  0.332199  0.211157  0.542590   
county_code                       0.051153 -0.172642 -0.095325 -0.180659   
site_num                         -0.009448 -0.026511 -0.003009 -0.034077   
parameter_code                         NaN       NaN       NaN       NaN   
poc                               0.017918  0.034916  0.013349  0.026793   
latitude                         -0.069728  0.420123  0.108789  0.208722   
longitude                         0.082722  0.109738  0.064940  0.153842   
observation_count                -0.018637  0.049663  0.002005 -0.021705   
observation_percent               0.055516  0.054192  0.072975  0.046256   
valid_day_count                   0.060305  0.037526  0.066153  0.039587   
required_day_count               -0.012961 -0.005597 -0.021111 -0.041370   
exceptional_data_count           -0.015671  0.022718 -0.003243  0.009862   
null_data_count                   0.036107  0.071198  0.058571  0.123971   
primary_exceedance_count               NaN       NaN       NaN       NaN   
secondary_exceedance_count             NaN       NaN       NaN       NaN   
num_obs_below_mdl                      NaN       NaN       NaN       NaN   
arithmetic_mean                  -0.097626 -0.041818 -0.097210 -0.220176   
arithmetic_standard_dev          -0.040919  0.066966  0.011779 -0.032616   
first_max_value                  -0.019674  0.042237  0.015727 -0.028967   
second_max_value                 -0.047990  0.045599 -0.013016 -0.074635   
third_max_value                  -0.065587  0.041254 -0.026346 -0.104638   
fourth_max_value                 -0.073682  0.038340 -0.032183 -0.120343   
first_max_non_overlapping_value        NaN       NaN       NaN       NaN   
second_max_non_overlapping_value       NaN       NaN       NaN       NaN   
ninety_nine_percentile           -0.016028  0.036134  0.015702 -0.028212   
ninety_eight_percentile          -0.037802  0.024572 -0.016424 -0.066386   
ninety_five_percentile           -0.072638  0.021920 -0.037743 -0.117673   
ninety_percentile                -0.098565  0.018565 -0.061254 -0.157967   
seventy_five_percentile          -0.101885 -0.018101 -0.092435 -0.209948   
fifty_percentile                 -0.104978 -0.071591 -0.122525 -0.256153   
ten_percentile                   -0.101591 -0.146060 -0.161285 -0.298754   

                                        3B        FP        HR        SO  \
year                             -0.313514  0.509662  0.473604  0.670212   
county_code                       0.108970  0.113688 -0.128150 -0.132404   
site_num                          0.018637  0.035352 -0.080180 -0.025134   
parameter_code                         NaN       NaN       NaN       NaN   
poc                              -0.059817  0.109179  0.020648  0.029693   
latitude                         -0.239368  0.094257  0.312006  0.077429   
longitude                        -0.195476  0.065803  0.161000  0.060094   
observation_count                -0.003768 -0.013450 -0.012209 -0.047196   
observation_percent               0.018705  0.031274  0.078042 -0.005630   
valid_day_count                   0.025305  0.020245  0.067403 -0.022130   
required_day_count                0.018460 -0.020726 -0.027170 -0.019426   
exceptional_data_count           -0.025315  0.056633  0.028072  0.070468   
null_data_count                  -0.093140  0.076911  0.114267  0.098577   
primary_exceedance_count               NaN       NaN       NaN       NaN   
secondary_exceedance_count             NaN       NaN       NaN       NaN   
num_obs_below_mdl                      NaN       NaN       NaN       NaN   
arithmetic_mean                   0.092940 -0.095416 -0.153899 -0.261471   
arithmetic_standard_dev           0.078558  0.006130 -0.019152 -0.096516   
first_max_value                   0.075705  0.014360 -0.020169 -0.101479   
second_max_value                  0.055384 -0.028604 -0.040956 -0.134544   
third_max_va

In [21]:

hydrocarbon_DF = merged_DF[merged_DF["parameter_name"] == "Total hydrocarbons"]
hydrocarbon_DF.corr().filter(focus).drop(focus)

W       ERA         H        2B  \
year                              0.348843  0.638573  0.468975  0.714955   
county_code                      -0.008440 -0.212650 -0.023692 -0.054407   
site_num                         -0.200017 -0.419188 -0.250634 -0.440461   
parameter_code                         NaN       NaN       NaN       NaN   
poc                               0.034965 -0.026217  0.013259 -0.016679   
latitude                          0.327146  0.526332  0.355357  0.585266   
longitude                         0.387553  0.331436  0.371175  0.550931   
observation_count                 0.022480  0.244042  0.119583  0.165218   
observation_percent               0.096591  0.298225  0.177938  0.241200   
valid_day_count                   0.028951  0.266407  0.124502  0.184664   
required_day_count               -0.106448  0.048424 -0.041964 -0.046244   
exceptional_data_count                 NaN       NaN       NaN       NaN   
null_data_count                   0.159514  0.151460  0.139229  0.273267   
primary_exceedance_count               NaN       NaN       NaN       NaN   
secondary_exceedance_count             NaN       NaN       NaN       NaN   
num_obs_below_mdl                      NaN       NaN       NaN       NaN   
arithmetic_mean                  -0.066726 -0.178011 -0.045915 -0.256159   
arithmetic_standard_dev           0.076715  0.038367  0.054832  0.034001   
first_max_value                   0.100016  0.103609  0.104178  0.106334   
second_max_value                  0.091171  0.088973  0.089851  0.087291   
third_max_value                   0.087481  0.079558  0.083020  0.075968   
fourth_max_value                  0.086617  0.076262  0.080835  0.072819   
first_max_non_overlapping_value        NaN       NaN       NaN       NaN   
second_max_non_overlapping_value       NaN       NaN       NaN       NaN   
ninety_nine_percentile            0.054916  0.024375  0.045521  0.003088   
ninety_eight_percentile           0.046180 -0.001287  0.035285 -0.028361   
ninety_five_percentile            0.018138 -0.072524  0.006889 -0.105588   
ninety_percentile                -0.017123 -0.142035 -0.023184 -0.185806   
seventy_five_percentile          -0.077274 -0.208789 -0.060308 -0.287372   
fifty_percentile                 -0.099963 -0.215860 -0.069019 -0.308966   
ten_percentile                   -0.101736 -0.184262 -0.060692 -0.272867   

                                        3B        FP        HR        SO  \
year                             -0.387803  0.617479  0.718751  0.730706   
county_code                       0.244557 -0.071249 -0.172560 -0.286232   
site_num                          0.103331 -0.368335 -0.379818 -0.273780   
parameter_code                         NaN       NaN       NaN       NaN   
poc                               0.120972  0.052204 -0.077014 -0.070162   
latitude                         -0.436171  0.492269  0.633429  0.395384   
longitude                        -0.238403  0.462370  0.512822  0.291218   
observation_count                -0.210742  0.057441  0.198275  0.216154   
observation_percent              -0.268812  0.166275  0.282675  0.261865   
valid_day_count                  -0.224987  0.077590  0.220203  0.244230   
required_day_count               -0.010864 -0.116707 -0.030020  0.049881   
exceptional_data_count                 NaN       NaN       NaN       NaN   
null_data_count                  -0.200124  0.281096  0.284967  0.341331   
primary_exceedance_count               NaN       NaN       NaN       NaN   
secondary_exceedance_count             NaN       NaN       NaN       NaN   
num_obs_below_mdl                      NaN       NaN       NaN       NaN   
arithmetic_mean                   0.030279 -0.253213 -0.154122 -0.025425   
arithmetic_standard_dev          -0.175180 -0.085150  0.142951  0.155960   
first_max_value                  -0.154445 -0.036878  0.193720  0.203221   
second_max_value                 -0.163755 -0.046601  0.182099  0.193395   
third_max_va

In [22]:
temperature_DF = merged_DF[merged_DF["parameter_name"] == "Average Ambient Temperature"]
temperature_DF.corr().filter(focus).drop(focus)

W       ERA         H        2B  \
year                             -0.059119 -0.409481 -0.380151 -0.254244   
county_code                      -0.088590  0.113118  0.082872  0.018097   
site_num                          0.032932 -0.152895 -0.085248 -0.165824   
parameter_code                         NaN       NaN       NaN       NaN   
poc                               0.046362 -0.135053 -0.042572 -0.047118   
latitude                          0.025471  0.084097  0.104672  0.116385   
longitude                         0.101749  0.050114  0.069509  0.188701   
observation_count                 0.041923 -0.007335  0.038887  0.039985   
observation_percent              -0.000549 -0.004544  0.022708  0.059093   
valid_day_count                   0.046279 -0.070743 -0.028140 -0.030910   
required_day_count                0.037432 -0.056331 -0.035666 -0.042160   
exceptional_data_count                 NaN       NaN       NaN       NaN   
null_data_count                   0.102997 -0.017868  0.129275  0.123387   
primary_exceedance_count               NaN       NaN       NaN       NaN   
secondary_exceedance_count             NaN       NaN       NaN       NaN   
num_obs_below_mdl                      NaN       NaN       NaN       NaN   
arithmetic_mean                   0.058148 -0.093902 -0.009390 -0.056451   
arithmetic_standard_dev          -0.059405  0.063920  0.008867  0.123233   
first_max_value                   0.039027 -0.065870  0.012079  0.039688   
second_max_value                  0.033320 -0.059424  0.009714  0.033540   
third_max_value                   0.024237 -0.039585  0.019237  0.047144   
fourth_max_value                  0.023792 -0.030543  0.012251  0.042817   
first_max_non_overlapping_value        NaN       NaN       NaN       NaN   
second_max_non_overlapping_value       NaN       NaN       NaN       NaN   
ninety_nine_percentile            0.024687 -0.062722  0.001234  0.030624   
ninety_eight_percentile           0.027152 -0.057938  0.002970  0.033903   
ninety_five_percentile            0.021181 -0.051530 -0.002478  0.027383   
ninety_percentile                 0.021085 -0.052637 -0.009572  0.022530   
seventy_five_percentile           0.020997 -0.059645 -0.013350  0.002008   
fifty_percentile                  0.053949 -0.073152  0.010213 -0.024823   
ten_percentile                    0.076289 -0.107402 -0.012421 -0.119011   

                                        3B        FP        HR        SO  \
year                             -0.008274  0.351142 -0.251561  0.561172   
county_code                      -0.001596 -0.003587 -0.062831 -0.180598   
site_num                          0.056122  0.046642 -0.155984 -0.000882   
parameter_code                         NaN       NaN       NaN       NaN   
poc                               0.008327  0.067267 -0.098677  0.025596   
latitude                         -0.063954  0.004745  0.139371 -0.014776   
longitude                        -0.115204 -0.007235  0.248080  0.048189   
observation_count                 0.032654  0.007694  0.010081 -0.075279   
observation_percent               0.036396 -0.053918 -0.071341 -0.110956   
valid_day_count                   0.003900  0.060530 -0.014655 -0.012143   
required_day_count               -0.009544  0.074823  0.015798  0.022778   
exceptional_data_count                 NaN       NaN       NaN       NaN   
null_data_count                   0.065166 -0.022521  0.021987 -0.074043   
primary_exceedance_count               NaN       NaN       NaN       NaN   
secondary_exceedance_count             NaN       NaN       NaN       NaN   
num_obs_below_mdl                      NaN       NaN       NaN       NaN   
arithmetic_mean                   0.034799  0.054241 -0.101794 -0.104345   
arithmetic_standard_dev          -0.017989 -0.058624  0.055413  0.013831   
first_max_value                   0.028786 -0.008731 -0.054999 -0.097180   
second_max_value                  0.027810  0.007317 -0.055207 -0.102545   
third_max_va

In [23]:
wind_DF = merged_DF[merged_DF["parameter_name"] == "Wind Speed - Resultant"]
wind_DF.corr().filter(focus).drop(focus)

W       ERA         H        2B  \
year                              0.000056  0.234971  0.100195  0.339287   
county_code                       0.042659 -0.103799  0.052561  0.098533   
site_num                         -0.023369 -0.017251 -0.037150 -0.040067   
parameter_code                         NaN       NaN       NaN       NaN   
poc                               0.059098 -0.070632 -0.003325 -0.006878   
latitude                          0.133576  0.131635  0.233970  0.123096   
longitude                         0.140278  0.045009  0.171607  0.150077   
observation_count                -0.048619 -0.042038 -0.028473 -0.023529   
observation_percent              -0.010092 -0.071975 -0.022081 -0.007411   
valid_day_count                  -0.043815 -0.025969 -0.003641 -0.020711   
required_day_count               -0.037935  0.015940  0.006110 -0.044162   
exceptional_data_count                 NaN       NaN       NaN       NaN   
null_data_count                   0.068937  0.028564  0.061904  0.082528   
primary_exceedance_count               NaN       NaN       NaN       NaN   
secondary_exceedance_count             NaN       NaN       NaN       NaN   
num_obs_below_mdl                      NaN       NaN       NaN       NaN   
arithmetic_mean                   0.082006  0.085595  0.105637  0.043506   
arithmetic_standard_dev           0.077296  0.060409  0.084730  0.021628   
first_max_value                   0.013764 -0.001304  0.007930  0.002219   
second_max_value                  0.101336  0.044222  0.108573  0.037125   
third_max_value                   0.099166  0.059615  0.118723  0.034208   
fourth_max_value                  0.098629  0.062187  0.118760  0.034440   
first_max_non_overlapping_value        NaN       NaN       NaN       NaN   
second_max_non_overlapping_value       NaN       NaN       NaN       NaN   
ninety_nine_percentile            0.087916  0.073660  0.102355  0.028004   
ninety_eight_percentile           0.084964  0.073387  0.098176  0.024002   
ninety_five_percentile            0.083407  0.075618  0.101486  0.033690   
ninety_percentile                 0.077768  0.077166  0.092639  0.030496   
seventy_five_percentile           0.077680  0.081511  0.097179  0.036900   
fifty_percentile                  0.081092  0.085152  0.107864  0.049432   
ten_percentile                    0.084239  0.099323  0.116237  0.058859   

                                        3B        FP        HR        SO  \
year                             -0.388662  0.604835  0.437874  0.724386   
county_code                       0.039169  0.021227 -0.086320 -0.042875   
site_num                         -0.039876  0.031592 -0.077326  0.025570   
parameter_code                         NaN       NaN       NaN       NaN   
poc                              -0.027162  0.071356  0.028698  0.072847   
latitude                         -0.047873  0.159724  0.287336  0.070723   
longitude                        -0.011611  0.094085  0.197288 -0.025838   
observation_count                -0.014996  0.009448 -0.099136  0.015178   
observation_percent              -0.030071  0.056856 -0.101349  0.005326   
valid_day_count                  -0.007809  0.012315 -0.080081  0.026036   
required_day_count                0.031092 -0.058356 -0.012544  0.010589   
exceptional_data_count                 NaN       NaN       NaN       NaN   
null_data_count                  -0.092191  0.056294  0.141635  0.032622   
primary_exceedance_count               NaN       NaN       NaN       NaN   
secondary_exceedance_count             NaN       NaN       NaN       NaN   
num_obs_below_mdl                      NaN       NaN       NaN       NaN   
arithmetic_mean                  -0.057334  0.009006  0.180364  0.037332   
arithmetic_standard_dev          -0.040426  0.006024  0.142962  0.020930   
first_max_value                  -0.042844  0.003245  0.039859  0.026420   
second_max_value                 -0.035333  0.042135  0.153311  0.015016   
third_max_va